src/routes/donations.js

```diff
@@ -154,6 +154,7 @@ router.post('/upload', (req, res) => {
           return previousPromise.then(_ => nextPromise(_))
         }, Promise.resolve([]))
       })
+      .then(_insertUploadRecord({ requestBody, t }))
```

Comment:
There will be an issue here. When the promise reaches here, whatever you put in the `then` will be *called*. So that needs to be a function. The way this is written, the `insertUploadRecord` function will be called out of sequence. Additionally, the next step in the promise chain needs the results, which is not being returned currently. So you need something like this:

```js
...
.then(results => {
  return _insertUploadFunction({ requestBody, t }).then(() => results)
})
```
This ensures two things:
1. The upload data is inserted at the correct time.
2. We pass on the results from the previous function to the next function in the promise chain.

Reply:
Thanks for the comprehensive explanation! However, without  the code you wrote, the response was still returned as usual, so I didnt realise anything wrong or different with the response.. Do you perhaps know why?

Reply:
Not 100% sure... it could be that the result got passed through anyway, but the insert upload would still be happening out of sequence.